### PR TITLE
Move loa_current out of User and into Session

### PR DESCRIPTION
**Pull request by omgitsbillryan**
- `User.loa.current` is now `Session.level`
- `User.loa.highest` is now `User.loa_highest`
- User now expires in Redis after 1 hour
- New route `/v0/sessions` will return `session.level`
- `@current_user.session` gets set in application_controller.rb
- All specs updated to have `@current_user.session` set

### CODE REVIEW

**Comment by omgitsbillryan** [comment]
@aub @saneshark I'm unable to get the spec to work, do you have any insight?
```/spec/services/mhv_logging_service_spec.rb:29```

**Comment by saneshark** [comment]
@omgitsbillryan it looks like you removed the generic :user factory. I would have add that back in as it is common practice to have a factory name mapped to the factory file name (singular). You can probably make it one of loa_user1 or loa_user3 I would think.

The specs are failing because there is no longer a factory having that name.

File: spec/factories/users.rb [hunk 1]
```diff
@@ -1,38 +1,21 @@
 # frozen_string_literal: true
 FactoryGirl.define do
-  factory :user, class: 'User' do
+  # this factory represents a "typical" LOA 1 user, in that, these
+  # are the attributes ID.me will return for LOA 1
+  factory :loa1_user, class: 'User' do
```
Comment (saneshark) [inline spec/factories/users.rb]: possibly consider keeping the factory name as user, and making loa1_user and loa2_user variations on the generic user that just set the specific attribute.

Comment (omgitsbillryan) [inline spec/factories/users.rb]: Ya, you're right about the convention, I will modify.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -80,6 +80,7 @@ def authenticate_token
         ::Digest::SHA256.hexdigest(@session.token)
       )
       @current_user = User.find(@session.uuid)
```
Comment (kreek) [inline app/controllers/application_controller.rb]: Could this and the next line be encapsulated in the User class, maybe a `User.from_session(session)` method? `@current_user.session = @session` pops up in a couple controllers.

File: app/controllers/v0/sessions_controller.rb [hunk 3]
```diff
@@ -27,26 +31,21 @@ def saml_callback
...
 
     private
 
-    def persist_session_and_user!
-      @session = Session.new(user_attributes.slice(:uuid))
-      @current_user = User.find(@session.uuid)
-      @current_user = saml_user if @current_user.nil? || up_level?
+    def persist_session_and_user
+      @session = Session.new(user_attributes.slice(:uuid).merge(level: loa_current))
+      @current_user = User.new(user_attributes)
```
Comment (kreek) [inline app/controllers/v0/sessions_controller.rb]: `User.from_session(session)` could be used here as well if you added `RedisStore.find_or_create` similar to `RedisStore.find_or_build`.

**Comment by omgitsbillryan** [comment]
The motivation for this PR is to fix this:
https://github.com/department-of-veterans-affairs/platform-team/issues/170

I'm having 2nd thoughts.  I started this work before @kreek did the work of separating Mvi data into its own redis store. Since that is done, we now have the option to sinply reduce the User lifetime from 24 hours down to 1 hour (I didn't have that option before Mvi was moved into its own Redis store since user was where Mvi data was stored).

This PR introduces a brittle connection between Session and User that I think is prone to bugs.  It also introduces API breaking changes, requiring F/E changes.

File: app/models/mvi.rb [hunk 4]
```diff
@@ -68,7 +69,7 @@ def select_source_id(correlation_id)
   end
 
   def mvi_response
-    return nil unless @user.loa3?
+    return nil unless user.session.loa3?
```
Comment (kreek) [inline app/models/mvi.rb]: If you wanted to keep it user.loa3? you could delegate to session in User.

**Comment by omgitsbillryan** [comment]
I'm closing this PR.

I've had 2nd thoughts - this PR introduces some brittle coupling between `Session` and `User` along with increased code complexity.  The real goal was to fix: https://github.com/department-of-veterans-affairs/platform-team/issues/170, so I've opened a new PR to take a more direct approach in fixing said bug (PR: https://github.com/department-of-veterans-affairs/vets-api/pull/573) that isnt as complex and will not require any F/E changes.

**Pull request closed without merging**